### ISSUE TITLE
The load of the users' credentials comes with  a confusing error message (Errno 2))

### ISSUE THREAD

**Issue by fab7** [cloudFPGA/cFSP#1]
Say I want to upload my credentials for a newly cloned repository, i.e. create a local `user.json` file. According to cFSP's documentation, I can can use the following command: 
```
    ./cfsp user load --username=<USERNAME> --password=<PASSWORD> --project=<PROJECT_NAME>
```
which produces the following output:
```
    0%|
    INFO: Repeat #0
    [Errno 2] No such file or directory: './user.json'
    Writing credentials template to ./user.json
    ['load']
```
I find the error message "**[Errno 2] No such file or directory**" confusing because the command successfully creates the `user.json` file and finishes without error.
Therefore, my suggestion is to remove or comment out line `120` of `cfsp_user.py`
``` 
    except Exception as e:
        # print(e)
        print("Writing credentials template to {}\n".format(json_file))
```

**Comment by diamantopoulos** [comment]
I completely agree that we should avoid an erroneous indication when a command is successful. 

But maybe commenting out the exception is neither ideal, since we want to print meaningful info when there is an error with the opening of the file (path issues, or syntax errors during the editing of the file, e.g. instead of `password` writing `passwd`).

We may have two options:

* commit a default user.json file to the repo, so that even the first time it doesn't throw an error.
* adding a command `cfsp user init` just for the first time.
 
What do you think @fab7 ?

**Comment by fab7** [comment]
Well, I think that you already know that the file `user.json` may not exist when enter the function `load_user_credentials` for the 1st time. Therefore, it is not an exception but it is expected. 
I would rather not handle the writing of the credentials in the same function that is used to read them, and create a dedicated function for that ... Or am I missing something?

**Comment by fab7** [comment]
Or rather add a write credential command (instead of re-using the ./cfsp user load) 
Something like : 
`./cfsp user write --username=YOUR_USERNAME  --password=YOUR_PASSWORD  --project=YOUR_PRPJECTNAME`

**Comment by diamantopoulos** [comment]
This is what is already supported, i.e. if you execute the command above, simply those credentials will be written in the file user.json. At that time I didn't want to have two commands, i.e. one for writing to the file and one for loading from the file since the latter could also write to the file. 

Why: There may be cases where you are the user of multiple projects and you just want to change this from the command line without again editing the credentials files. Then you just execute 

`./cfsp user load --username=YOUR_USERNAME --password=YOUR_PASSWORD --project=YOUR_PRPJECTNAME_1`

`./cfsp user load --username=YOUR_USERNAME --password=YOUR_PASSWORD --project=YOUR_PRPJECTNAME_2`

and the `user.json` will be updated (written) accordingly.

Then I would avoid the "write". But what about the "init":

e.g. just for the first time:

`./cfsp user init --username=YOUR_USERNAME --password=YOUR_PASSWORD --project=YOUR_PRPJECTNAME_1`

then a user.json file will be created with the credentials above. If the user doesn't specify those, a default file will be created. e.g.

`./cfsp user init`

Then any following call of `cfsp user load` will not be erroneous and we'll keep the exception printing (which is useful if sth goes wrong). What do you think?

**Comment by fab7** [comment]
>Then I would avoid the "write". But what about the "init":
I am fine with 'init', also a good idea. 
1) I just don't want the user to receive an `errno` the very 1st time he tries to add his credentials, while he does everything right and everything went right ;-) 
2) I also don't understand why you would create a default `user.json` ? Either the user provides the 3 parameters and you create the file, or he does not and the command should fail (but this is more of a detail)

**Comment by diamantopoulos** [comment]
1. Fully agree.
2. The arguments username, username, and project are optional in `docopt`. I wanted to have them optional so that a user can change just one of them (e.g. project) without the necessity to provide them all the time. But of course, we can do it as you say. Let me proceed, thx

**Comment by diamantopoulos** [comment]
ok fixed it in: ed33a57aff3587c64dd0a2d55f44ea566e5242cd

...and updated documentation in the previous and in the: 20122a7136054f8780cd46edc71b62e33bef9153 .

You just need to run the following and no errors will be raised :D  `./cfsp user load  --username=my_username --password=my_password --project=my_project` 

If it is fine you may close the issue @fab7 . Thanks for bringing it up!
